pkg/network/multus/annotations.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package multus
+
+import (
+	"encoding/json"
+	"fmt"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	k8sv1 "k8s.io/api/core/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+func IterateMultusNonDefaultNetworks(vmi *v1.VirtualMachineInstance, fn func(network v1.Network, networkRequestInterfaceName string)) {
```

Comment:
per of my comment on the spec this need to be change to support all networks. 
It better to look if network has k8s.v1.cni.cncf.io/resourceName in the annotation

Reply:
Hey @moshe010, do you mean the `k8s.v1.cni.cncf.io/resourceName` annotation on the NetworkAttachmentDefinition?
If so then I'm afraid it is not enough to directly map between the network and the pci-address.
Consider a situation where you have 2 networks (with different MACs) connected to the same NetworkAttachmentDefinition? in that case both will have the same k8s.v1.cni.cncf.io/resourceName (more info [here](https://github.com/kubevirt/community/pull/185/files#diff-aee927941b6b876c1ac6b0f0a74cac867f8112545776978799c0c065925f2c2eR285)).

We chose to rely on the multus pod annotation: `k8s.v1.cni.cncf.io/networks-status`. In it - you have a 1:1 mapping between the network name and the host-pci-address
To to the mapping we need to understand a few things:
1.  in the the `k8s.v1.cni.cncf.io/networks-status`, each network is distinguished by the`interface` field.
2.  The`interface` field is set from `k8s.v1.cni.cncf.io/networks` (which is created by kubevirt when creating the pod) 

`IterateMultusNonDefaultNetworks` is the function that makes sure that we give every network in vmi.spec.networks the appropriate and **consistent** name - so that we can rely on it when mapping the SRIOV networks via the `k8s.v1.cni.cncf.io/networks-status`

Reply:
when we use ovn-kubernetes with ovs hardware offload we use multus and NAD for primary network (even if there is not secanday network NAD)
 
if I understand you current logic,  you iterate on no default NAD (which mean secondary network). My comment was that you need to iterate also on the primary network which can be (NAD)  and handle it. 
update 

	for i, network := range multusNonDefaultNetworks {
                if network.default:
                   networkRequestInterfaceName := "eth0"
               else:
		   networkRequestInterfaceName := fmt.Sprintf("net%d", i+1)
		fn(network, networkRequestInterfaceName)
	}

and remove the filterMultusNonDefaultNetworks 

my comment regarding "k8s.v1.cni.cncf.io/resourceName" was incorrect because I don't understand what the code does.

Reply:
I see. can you paste a vmi and NAD manifests with the scenario you are referring to? it would make it easier for me to study and include this case.

Reply:
I am so sorry. I verified gain and we don't use primary network with kubevirt only with Pod. 
please discard all my comments on the support of primary network in kubevirt. your logic is good to assume secondary network only. Also please discard this from the spec proposal. 
Again sorry for the confusion

Reply:
> when we use ovn-kubernetes with ovs hardware offload we use multus and NAD for primary network (even if there is not secanday network NAD)

@moshe010 , you are using the primary as a network device or as a vfio device?
At the moment we do not have this scenario covered in out e2e tests, and we do not map the primary network to a resource pool in the current implementation.
Said all that, I do not see the a reason not to add it to the mapping as well, in practice "fixing" an older limitation. But we better do it in a separate PR, one that we can reason for it with examples of usages.

Reply:
np @moshe010, thanks for letting me know. 
In any case, should the need for this arise in the future, it would be fairly easy to also map the primary network (but I agree that it should happen on a different PR).

Reply:
@EdDev @RamLavi 
So it seem there are 2 use case for VF on primary network:
1. VF as vfio device to pass though to the VM. currently it not support in kubevirt
2. VF as net-device which moved to Pod network namespace. The VF is connected to linux bridge and on the other end VM tap device. (this case is supported in kubevirt)
So for case 2 we still need to support primary network in the current code.